jupyterhub/spawners.py

```diff
@@ -144,17 +144,21 @@ def start(self, *args, **kwargs):
                 )
             )
 
-            if (
-                len(access_levels) > 0
-                and max(access_levels) >= gitlab.MAINTAINER_ACCESS
-            ):
+            access_level = gitlab.GUEST_ACCESS
+            if len(access_levels) > 0:
+                access_level = max(access_levels)
+
+            if access_level >= gitlab.MAINTAINER_ACCESS:
 
                 environment = {
                     variable.key: variable.value
                     for variable in gl_project.variables.list()
                 }
                 self.environment.update(environment)
 
+            if access_level >= gitlab.DEVELOPER_ACCESS:
+                self.environment["GITLAB_AUTOSAVE"] = "1"
```

Comment:
This will only add it to the environment variables of the notebook container. You have to explicitly add it to the env variables of the init container too (see line 218).

Reply:
Thanks @ableuler for noticing this! It's been a while that I've touched this code base!

Reply:
No problem. I spent many of my quarantine days staring at this, so I knew what to look out for... 😂